### DISCUSSION TITLE
A Verbose or Debug output?

### DISCUSSION THREAD

**Discussion by jtayler** [mat/besticon#69] [Q&A]
This is a great solution, but I find it doesn't work for a variety of sites and I can't see why not. Is there any way to get a trace or verbose output or debug output so we can see how to fix or figure sites that do not work?

**Comment by brleinad** [comment]
This would be very beneficial for my team too. We would like to track what are the most requested icons and which ones can we not find. I would be happy to open a PR for this.

**Comment by mat** [comment]
Sounds like a useful thing to have. Maybe easiest to start with debug output in the form of the 'check results' like

```
Checking "http://example.com/favicon.ico": not present
Getting "http://example.com/" contents: ok
...
```

would be nice and good also be used to return this as part of or instead of the response to a 'failed' request.

**Reply by brleinad** [reply]
That sounds good. I can start by just adding the debug output for now. Should there be a config variable to optionally avoid having this output? 

Maybe just show the debug output if `DEBUG_OUTPUT=true`.

**Reply by mat** [reply]
Sounds good to me 👍
